elastic/integrations
# [Kubernetes] Add state namespace datastream

## What

Add state namespace data stream.

The metricset was added in version 8.11.0.

Dimensions for TSDB:
- `kubernetes.namespace` is unique, so it is a dimension.
- `agent.id` is a common dimension among data streams.
- `orchestrator.cluster.url` is a dimension for all kubernetes data streams.

<details>

<summary>Results from running TSDB testing kit:</summary>

```
The time series fields for the TSDB index are: 
	- dimension (3 fields):
		- agent.id
		- kubernetes.namespace
		- orchestrator.cluster.url
	- gauge (1 fields):
		- kubernetes.state_namespace.created.sec
	- routing_path (3 fields):
		- agent.id
		- kubernetes.namespace
		- orchestrator.cluster.url

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-kubernetes.state_namespace-default-2023.10.30-000001 to tsdb-index-enabled...
All 210 documents taken from index .ds-metrics-kubernetes.state_namespace-default-2023.10.30-000001 were successfully placed to index tsdb-index-enabled.
```
</details>

## Checklist



## How to test this PR locally

1. Clone this branch.
2. Build the package.
3. Update the service registry.
4. Add Kubernetes to a policy and check the new data stream.

## Related issues

Relates to https://github.com/elastic/elastic-agent/issues/3100.

## Screenshots

In Discover:
![image](https://github.com/elastic/integrations/assets/113898685/4ade89ea-376e-47f8-bec3-5300ea6a6dc8)

In the integration itself:
![image](https://github.com/elastic/integrations/assets/113898685/b574a51c-ed66-4afe-ad7e-9b391a7fdbeb)
